### PR TITLE
API error return converged to one func

### PR DESCRIPTION
**Pull request by swapnili**
Use ERROR_INT() or ERROR_PTR() to return errors from API's.
This way if we want to change the behaviour later, we will just need
to update these funcitons.
Also fixed some error return cases and comments.

Signed-off-by: Swapnil Ingle <swapnil.ingle@nutanix.com>

### CODE REVIEW

File: lib/irq.c [hunk 1]
```diff
@@ -403,14 +400,12 @@ vfu_irq_trigger(vfu_ctx_t *vfu_ctx, uint32_t subindex)
 
     ret = validate_irq_subindex(vfu_ctx, subindex);
     if (ret < 0) {
-        return ret;
+        return ERROR_INT(-ret);
```
Comment (jlevon) [inline lib/irq.c:403]: should be ERROR_INT(EINVAL)

File: lib/irq.c [hunk 2]
```diff
@@ -380,15 +380,12 @@ static int
 validate_irq_subindex(vfu_ctx_t *vfu_ctx, uint32_t subindex)
```
Comment (jlevon) [inline lib/irq.c:380]: now this doesn't fiddle with errno, can you make it a boolean return like other validate functions?

**Review by jlevon — CHANGES_REQUESTED** [review]
Thanks for this, nice cleanup.

File: lib/irq.c (hunk 1 above)
Comment (swapnili) [reply lib/irq.c]: Thanks. Fixed it.

File: lib/irq.c (hunk 2 above)
Comment (swapnili) [reply lib/irq.c:380]: Sure

File: lib/irq.c [hunk 3]
```diff
@@ -376,41 +376,34 @@ handle_device_set_irqs(vfu_ctx_t *vfu_ctx, uint32_t size,
...
 int
 vfu_irq_trigger(vfu_ctx_t *vfu_ctx, uint32_t subindex)
 {
-    int ret;
     eventfd_t val = 1;
 
-    ret = validate_irq_subindex(vfu_ctx, subindex);
-    if (ret < 0) {
-        return ret;
+    if (validate_irq_subindex(vfu_ctx, subindex) == false) {
```
Comment (jlevon) [inline lib/irq.c:400]: nit, sorry, but don't ever do "== <boolean>", this should just be:

  if (!validate_irq_subindex()) {

File: lib/irq.c [hunk 4]
```diff
@@ -423,8 +416,8 @@ vfu_irq_message(vfu_ctx_t *vfu_ctx, uint32_t subindex)
     struct vfio_user_irq_info irq_info;
 
     ret = validate_irq_subindex(vfu_ctx, subindex);
-    if (ret < 0) {
-        return -1;
+    if (ret == false) {
```
Comment (jlevon) [inline lib/irq.c:419]: ditto:

if (!validate_irq_subindex()) {

Here, we also shouldn't set ret, as that's an int

**Review by jlevon — CHANGES_REQUESTED** [review]
w

File: lib/irq.c (hunk 3 above)
Comment (swapnili) [reply lib/irq.c]: Yes I also do not like doing "==", but its the frodo coding_style which says so at-least for checking NULL.
Changed it to !

File: lib/irq.c (hunk 4 above)
Comment (swapnili) [reply lib/irq.c]: Done

**Review by jlevon — APPROVED** [review]

**Pull request merged**